examples/js-sub.c

```diff
@@ -130,21 +175,50 @@ int main(int argc, char **argv)
 
     if (s == NATS_OK)
     {
-        if (pull)
+        if (pull && async)
+        {
+            jsOpts.PullSubscribeAsync.FetchSize = 17;
+            jsOpts.PullSubscribeAsync.KeepAhead = 7;
+            jsOpts.PullSubscribeAsync.CompleteHandler = _completeFetchCb;
+
+            jsFetchRequest lifetime;
```

Comment:
@kozlovic @Jarema @mtmk I chose to have a separate `lifetime` parameter as a shortcut for passing these options. 2/5 I am considering removing the extra parameter, and adding all these values to `jsOpts.PullSubscribeAsync`. (a) It would be more idiomatic; (b) arguably, CompleteHandler a more common use-case and they should be "level"-ed; and (c) the use of `jsFetchRequest` struct may be confusing since no such request will ever be made.

Reply:
You could. The thing I like about the options, is if we decide down the line that we need other option(s), then the function definition stays the same. We don't have to come up with a different function name, etc..
That being said, you could pass the parameters that seem essential at this time, and if in the future we need more, then add them through the options?

Reply:
Replaced `lifetime` with more `jsOpts`, looks cleaner.

Reply:
@Jarema @mtmk Can I please get your review on the API iteself (parameters, etc.) before we finalize this?